src/ui/public/directives/kbn_top_nav.js

```diff
@@ -79,22 +81,34 @@ module.directive('kbnTopNav', function (Private) {
         } else {
           ctrlObj.curr = which;
         }
+
         const templateToCompile = ctrlObj.templates[ctrlObj.curr] || false;
-        $scope.kbnTopNav.currTemplate = templateToCompile ? $compile(templateToCompile)($scope) : false;
+
+        if ($scope.kbnTopNav.currTemplate) {
+          $scope.kbnTopNav.currTemplateScope.$destroy();
+          $scope.kbnTopNav.currTemplate.remove();
+        }
+
+        if (templateToCompile) {
+          $scope.kbnTopNav.currTemplateScope = $scope.$new();
```

Comment:
So that directives are properly torn down. The welcome directive used in the console "welcome" drop-down uses `$scope.$on('$destroy', ...);` to do cleanup logic, but since the scope is never destroyed the cleanup never happens. This is why ng-if creates a child scope, because parts of the dom need to have their own scope when they are removed for cleanup purposes.